File: pet_hospital_backend/src/main/java/com/pethospital/entity/ServiceItem.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;

@Data
@TableName("services")
public class ServiceItem {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private String name;
    
    /**
     * 类别: VACCINE, SURGERY, EXAM, LAB
     */
    private String category;
    
    private BigDecimal price;
    
    private String description;
    
    private Integer isActive;
}

File: pet_hospital_frontend/src/api/serviceItem.ts
import api from './index'
import type { PageResult } from './customer'

export interface ServiceItem {
    id?: number
    name: string
    category: string // VACCINE, SURGERY, EXAM, LAB
    price: number
    description?: string
    isActive?: number
}

export const serviceItemApi = {
    list: (params: { page?: number; size?: number }) =>
        api.get<any, { code: number; data: PageResult<ServiceItem> }>('/services', { params }),

    listActive: () =>
        api.get<any, { code: number; data: ServiceItem[] }>('/services/active'),

    create: (data: ServiceItem) =>
        api.post<any, { code: number; data: boolean }>('/services', data),

    update: (id: number, data: ServiceItem) =>
        api.put<any, { code: number; data: boolean }>(`/services/${id}`, data),

    delete: (id: number) =>
        api.delete<any, { code: number; data: boolean }>(`/services/${id}`)
}
